File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

float get_change(void);

int main(void)
{
    //types of coins
    int quarters = 0;
    int dimes = 0;
    int nickels = 0;
    int pennies = 0;
    //get the change due and round it to cents
    float change = get_change();
    int cents = round(change * 100);
    //take off quaters
    while (cents >= 25)
    {
        cents -= 25;
        quarters ++;
    }
    //take off dimes
    while (cents >= 10)
    {
        cents -= 10;
        dimes ++;
    }
    //take off nickels
    while (cents >= 5)
    {
        cents -= 5;
        nickels ++;
    }
    //take off pennies
    while (cents >= 1)
    {
        cents -= 1;
        pennies ++;
    }
    //add the number of each coin used
    int coins = quarters + dimes + nickels + pennies;
    printf("%i\n", coins);


}
float get_change(void)
{
    float change;
    do
    {
        change = get_float("Change Owed: ");
    }
    while (change < 0);
    return change;
}

File: pset6/cash/cash.py
from cs50 import get_float


def main():
    coins = 0
    change = get_change()

    # quarters
    while change >= .25:
        change -= .25
        coins += 1
        change = round(change, 2)

    # dimes
    while change >= .10:
        change -= .10
        coins += 1
        change = round(change, 2)

    # nickles
    while change >= .05:
        change -= .05
        coins += 1
        change = round(change, 2)

     # pennies
    while change >= .01:
        change -= .01
        coins += 1
        change = round(change, 2)

    print(coins)


def get_change():
    # get valid change
    change = get_float("Change owed: ")
    while change < 0:
        change = get_float("Change owed: ")
    # round
    change = round(change, 2)
    return change


main()